### PR TITLE
Made catch error cog and make cog management system.

### PR DESCRIPTION
**Pull request by TNTprizz**

### CODE REVIEW

**Comment by windowsboy111** [comment]
I feel like we need to rename config.yml to config.yml.example or sth similar

File: src/bot.py [hunk 1]
```diff
@@ -6,31 +6,37 @@
...
+    print("DEBUG: Using external prefix " + cmdarg[2])
 
-# For changing prefix
-async def determine_prefix(bot, message):
-  guild = message.guild # Get the guild so that cannot change prefix in DMChannel
-  if guild: # If it is not the DMChannel
-    return bot.custom_prefixes.get(guild.id, default_prefixes) # Get prefix from the bot.custom_prefixes variable
-  else: # If it is a DMChannel
-    return default_prefixes # Return the default prefix
+intents = discord.Intents().all()
```
Comment (windowsboy111) [inline src/bot.py:26]: `discord.Intents.all()`

File: src/cogs/Bash.py [hunk 2]
```diff
@@ -48,29 +52,84 @@ async def ping(self, ctx):
...
+            if filename.endswith('.py'):
+                status = ""
+                try:
+                    self.bot.load_extension("cogs." + filename[:-3])
+                    del self.bot.cmdlist[filename[:-3]]
+                    self.bot.unload_extension("cogs." + filename[:-3])
+                    status = "Inactive"
+                except commands.ExtensionAlreadyLoaded:
+                    status = "Active"
+                except:
```
Comment (windowsboy111) [inline src/cogs/Bash.py:92]: maybe `except Exception:`

**Comment by TNTprizz** [comment]
Winboi:
> 
> I feel like we need to rename config.yml to config.yml.example or sth similar

No need cuz no one knows that touching `config.yml` is essential for the bot to run
just config.yml ok, no need example

File: src/cogs/Vote.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                ""
+            content:
+                ""
+        }
+}
+"""
+import discord
+from discord.ext import commands
+
+mvotelist = ["🇦","🇧","🇨","🇩","🇪","🇫","🇬","🇭","🇮","🇯","🇰","🇱","🇲","🇳","🇴","🇵","🇶","🇷","🇸","🇹"]
```
Comment (windowsboy111) [inline src/cogs/Vote.py:28]: make them a str, not list. see `wb/patch`.
Same goes for `tfvotelist`.

File: src/cogs/Vote.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            #except:
+                #raise commands.BadArgument("'" + str(messageid) + "' is not a valid message id")
+        for r in msg.reactions:
+            await msg.clear_reaction(r.emoji)
+        await msg.add_reaction("🛑")
+        await msg.reply(embed = emb)
+
+def setup(bot):
+    bot.tfvote = {}
+    bot.mvote = {}
```
Comment (windowsboy111) [inline src/cogs/Vote.py:122]: See `wb/patch`

File: src/cogs/error.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+from discord.ext import commands
+import asyncio
+
+class Error(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @commands.Cog.listener()
+    async def on_command_error(self, ctx, error):
+        if isinstance(error, commands.MissingPermissions) or isinstance(error, commands.MissingRole) or isinstance(error, commands.MissingAnyRole):
```
Comment (windowsboy111) [inline src/cogs/error.py:11]: `if isinstance(error, (commands.MissingPermissions, commands.MissingRole, commands.MissingAnyRole)):`

**Review by windowsboy111 — CHANGES_REQUESTED** [review]

File: src/bot.py (hunk 1 above)
Comment (TNTprizz) [reply src/bot.py:26]: don't work if no Intents`()`

File: src/cogs/Bash.py (hunk 2 above)
Comment (TNTprizz) [reply src/cogs/Bash.py:92]: same la but I will change it

File: src/cogs/Vote.py (hunk 3 above)
Comment (TNTprizz) [reply src/cogs/Vote.py:28]: ahh ok

**Comment by windowsboy111** [comment]
1. Understand that cogs and extensions are different.
2. you don't always need `bot`.
3. pro tip: `contextlib.suppress()`
4. strings can be indexed.

File: src/bot.py (hunk 1 above)
Comment (TNTprizz) [reply src/bot.py:26]: whoops sor

File: src/cogs/Vote.py (hunk 4 above)
Comment (TNTprizz) [reply src/cogs/Vote.py:122]: what is in `wb/patch`

**Comment by TNTprizz** [comment]
> 
> 
>     1. Understand that cogs and extensions are different.
> 
>     2. you don't always need `bot`.
> 
>     3. pro tip: `contextlib.suppress()`
> 
>     4. strings can be indexed.

1. So, what we need is a extension directory
2. My goal is to reach the goal
3. Pro tip: I am noob
4. This is right, but the regional indicator sticks together when not using list and that is...... just annoying

**Comment by windowsboy111** [comment]
Merged with #8.

**Pull request closed without merging**